pennylane/grouping/group_observables.py

```diff
@@ -191,7 +192,7 @@ def group_observables(observables, coefficients=None, grouping_type="qwc", metho
     Args:
         observables (list[Observable]): a list of Pauli word ``Observable`` instances (Pauli
             operation instances and :class:`~.Tensor` instances thereof)
-        coefficients (list[float]): A list of float coefficients. If not specified,
+        coefficients (tensor_like): A tensor of coefficients. If not specified,
```

Comment:
This PR makes the transform interface-agnostic. The next PR adds grouping to the Hamiltonian class.